Fix outlined button stroke defaults

clocksmith opened the pull request:
Primarily, this change corrects the stroke defaults to color: black 12% alpha and width: 1dp

This change also:
1. Extracts the resolver for outline color into its own function. This type of logic will most likely be consolidated as we improve the theming library.
2. Adds unit tests for the newly affected disabled state, and re-organizes the existing test so that it clearly tests each state for the component specific overrides.
3. Leaves various TODOs for properly honoring the buttonThemeData and themeData constants that do not yet exist.

* I think the dartfmt was correct in these instances, but I apologize in advance if it did the wrong thing.

closes #18170

clocksmith committed:
fixed stroke size and stroke color

clocksmith committed:
fixed stroke size and stroke color

clocksmith committed:
dartfmt did a lot here, but updated the outline color

clocksmith committed:
dartfmt did a lot here, but updated the outline color

clocksmith committed:
fixed

clocksmith committed:
fixed

clocksmith committed:
undo dartfmt for lines not touched and use existing nest structure for assigning color

clocksmith committed:
undo dartfmt for lines not touched and use existing nest structure for assigning color

clocksmith committed:
Fixed docs and formatting

clocksmith committed:
Fixed docs and formatting

clocksmith committed:
removed docs and even further simpified color assignment

clocksmith committed:
removed docs and even further simpified color assignment

clocksmith committed:
trying to get original test to pass

clocksmith committed:
trying to get original test to pass

clocksmith committed:
revert outline button test

clocksmith committed:
revert outline button test

clocksmith committed:
refactor

clocksmith committed:
refactor

clocksmith committed:
Adjusted color rules to current spec and refactored color extraction into its own function, still working on tests

clocksmith committed:
Adjusted color rules to current spec and refactored color extraction into its own function, still working on tests

clocksmith committed:
space fix

clocksmith committed:
space fix

clocksmith committed:
update

clocksmith committed:
update

clocksmith committed:
m

clocksmith committed:
m

clocksmith committed:
Updated outline button files

clocksmith committed:
Updated outline button files

clocksmith committed:
Fix comment

clocksmith committed:
Fix comment

clocksmith committed:
moved 0.12 opacity to constant

clocksmith committed:
moved 0.12 opacity to constant

clocksmith commented on packages/flutter/lib/src/material/outline_button.dart:
```diff
@@ -354,22 +354,31 @@ class _OutlineButtonState extends State<OutlineButton> with SingleTickerProvider
+      // TODO(clocksmith): Check buttonThemeData once its passed.
+      final Color themeColor = theme.primaryColor ??
+          theme.brightness == Brightness.dark ? Colors.white : Colors.black;
+      return widget.borderSide?.color ?? themeColor.withOpacity(0.12);
```
Done

clocksmith committed:
Fix ?? ordering

clocksmith committed:
Fix ?? ordering

clocksmith commented:
This is good to go (minus the current flutter-build being broken)

Hixie commented on packages/flutter/lib/src/material/outline_button.dart:
```diff
@@ -349,27 +351,37 @@ class _OutlineButtonState extends State<OutlineButton> with SingleTickerProvider
+      // TODO(clocksmith): Check theme.surfaceColor once its available.
+      // TODO(clocksmith): Check buttonThemeData once its passed.
+      final Color themeColor = theme.primaryColor ??
+          (theme.brightness == Brightness.dark ? Colors.white : Colors.black);
```
use a switch here (see Flutter style guide)

clocksmith replied:
done

Hixie commented on packages/flutter/test/material/outline_button_test.dart:
```diff
@@ -26,9 +27,12 @@ void main() {
     );
-    expect(tester.widget<OutlineButton>(find.byType(OutlineButton)).enabled, true);
+    expect(
+        tester.widget<OutlineButton>(find.byType(OutlineButton)).enabled, true);
```
please don't indent like this; see style guide

clocksmith replied:
done

Hixie commented on packages/flutter/test/material/outline_button_test.dart:
```diff
@@ -223,26 +260,31 @@ void main() {
+  testWidgets('OutlineButton implements debugFillDescription',
+      (WidgetTester tester) async {
+    final DiagnosticPropertiesBuilder builder =
+        new DiagnosticPropertiesBuilder();
```
these are more readable on one line like before

clocksmith replied:
done

Hixie commented on packages/flutter/test/material/outline_button_test.dart:
```diff
@@ -43,121 +47,152 @@ void main() {
-    ));
+          ),
+          ignoreId: true,
+        ));
```
why this change? it doesn't seem more readable this way.

maybe dartfmt got confused because there's no trailing commas?

clocksmith replied:
I've removed dartfmt

Hixie reviewed (commented):
if you want to use dartfmt, please increase the line width to avoid egregious line breaking, and be vigilant about trailing commas. 0

Hixie commented:
i don't really understand #18170; incorrect according to what?

cc @HansMuller

clocksmith committed:
fix formatting

clocksmith committed:
fix formatting

clocksmith committed:
fix formatting

clocksmith committed:
fix formatting

clocksmith committed:
fix formatting

clocksmith committed:
fix formatting

clocksmith committed:
fix formatting

clocksmith committed:
fix formatting

clocksmith committed:
fix formatting

clocksmith committed:
fix formatting

clocksmith commented:
Comments are addressed, PTAL

cbracken committed:
Revert "Add segmented control to gallery (#19192)" (#19227)

This appears to trigger an error during AOT snapshot of the gallery as
follows:

```
stdout: [   +6 ms] Building AOT snapshot in release mode (ios-release)...
stdout: [  +15 ms] Extra gen_snapshot options: [--print_snapshot_sizes]
stdout: [   +7 ms] /usr/bin/arch -x86_64 /Users/flutter/.cocoon/flutter/bin/cache/artifacts/engine/ios-release/gen_snapshot --await_is_keyword --causal_async_stacks --packages=.packages --url_mapping=dart:ui,/Users/flutter/.cocoon/flutter/bin/cache/pkg/sky_engine/lib/ui/ui.dart --url_mapping=dart:vmservice_io,/Users/flutter/.cocoon/flutter/bin/cache/pkg/sky_engine/sdk_ext/vmservice_io.dart --embedder_entry_points_manifest=/Users/flutter/.cocoon/flutter/bin/cache/artifacts/engine/ios-release/dart_vm_entry_points.txt --embedder_entry_points_manifest=/Users/flutter/.cocoon/flutter/bin/cache/artifacts/engine/ios-release/dart_io_entries.txt --print_snapshot_sizes --snapshot_kind=app-aot-assembly --assembly=build/aot/arm64/snapshot_assembly.S /Users/flutter/.cocoon/flutter/examples/flutter_gallery/lib/main.dart
stderr: [+9324 ms] Error: 'file:///Users/flutter/.cocoon/flutter/examples/flutter_gallery/lib/demo/cupertino/cupertino_segmented_control_demo.dart': error: line 19 pos 1: circular dependency for function _CupertinoSegmentedControlDemoState
stderr: [        ] class _CupertinoSegmentedControlDemoState extends State {
stderr: [        ] ^
stderr: [  +33 ms] Dart snapshot generator failed with exit code 254
stderr: [        ] Snapshotting (IOSArch.arm64) exited with non-zero exit code: 254
stdout: [        ] Built to build/aot/.
stdout: [  +10 ms] "flutter aot" took 9,485ms.
```

This reverts commit 9d49ee3bf06e634e53331b544dbb640453f81348.

cbracken committed:
Roll engine to ce06bbac58d7c68487dd09616864c904e2f71a72 (#19213)

* ensure a11y state is communicated back to flutter (flutter/engine#5631)
* Roll src/third_party/skia b1b87d9df81e..a1e5630183c1 (10 commits) (flutter/engine#5637)
* Roll src/third_party/skia 385804514edf..b1b87d9df81e (6 commits) (flutter/engine#5636)
* Roll src/third_party/skia e97bb26893a1..385804514edf (1 commits) (flutter/engine#5635)
* Roll src/third_party/skia e2e52e46ca63..e97bb26893a1 (1 commits) (flutter/engine#5634)
* Roll src/third_party/skia 14de25dfc7c4..e2e52e46ca63 (2 commits) (flutter/engine#5633)
* Roll src/third_party/skia 551dc3e91143..14de25dfc7c4 (2 commits) (flutter/engine#5630)
* libtxt: apply an offset to drawing operations instead of translating the canvas (flutter/engine#5628)
* Roll src/third_party/skia 059a9ab4bcd0..551dc3e91143 (5 commits) (flutter/engine#5629)
* Roll src/third_party/skia 45c9dab4c3ec..059a9ab4bcd0 (8 commits) (flutter/engine#5627)
* Remove some unused code from the Android host (flutter/engine#5619)
* Roll src/third_party/skia 37b7e4714558..45c9dab4c3ec (1 commits) (flutter/engine#5626)
* Roll src/third_party/skia c421ca1d6e41..37b7e4714558 (2 commits) (flutter/engine#5624)
* Roll src/third_party/skia 5b201e3b0ba8..c421ca1d6e41 (3 commits) (flutter/engine#5623)
* Roll src/third_party/skia d47fe095cf88..5b201e3b0ba8 (6 commits) (flutter/engine#5622)
* Roll src/third_party/skia 9c0ce41cf711..d47fe095cf88 (12 commits) (flutter/engine#5618
* Roll src/third_party/skia d4b2adeaa929..9c0ce41cf711 (7 commits) (flutter/engine#5617)
* Roll src/third_party/skia f46710802ad5..d4b2adeaa929 (1 commits) (flutter/engine#5616)

liyuqian committed:
Increase image cache to 100MB (#19220)

This should fix #18952 and #813

cbracken committed:
Roll engine to 4466d61a993fdbde2cb092b4da9bbef33d1962fe (#19230)

Includes:
* Remove vmservice_io.main from entry points. (flutter/engine#5625)
* Roll src/third_party/skia 7e2327b133db..723b1f6ef941 (1 commits) (flutter/engine#5642)
* Roll src/third_party/skia 24d18ced1ad7..7e2327b133db (6 commits) (flutter/engine#5641)
* Roll src/third_party/skia a1e5630183c1..24d18ced1ad7 (7 commits) (flutter/engine#5639)

Hixie committed:
Remove hacks that make macOs, Linux, and Windows not fail. (#19187)

They should fail, because we don't have real support yet.

Also, make the debug override work in release builds, so that people
on those platforms have a workaround.

cbracken committed:
Revert "Roll engine to 4466d61a993fdbde2cb092b4da9bbef33d1962fe (#192… (#19233)

…30)"

Causes microbenchmarks to start failing. This included the following
engine PR, which appears to be the culprit:
* Remove vmservice_io.main from entry points. (flutter/engine#5625)

This reverts commit 3643790437f36a11e4ed19f76a1865ff3b0f3a4d.

aam committed:
Fix 'are there any views available' check (#19231)

gspencergoog committed:
Make HotRunner call pub get automatically if the pubspec is out of date. (#19183)

This adds a call to 'pub get' if the pubspec.yaml is out of date wrt the .packages file when a hot reload is requested.

Fixes #15879

DanTup committed:
Remove note about port being removed (#19263)

Some things require a port (eg. `flutter attach`) and will forward it on to the device, so we shouldn't force clients to parse it out of a uri.

shrike69 committed:
Use binary name to locate .packages file on Fuchsia (#19217)

goderbauer committed:
Do not assert when semantics are on and app draws on 0x0 surface (#19059)

Instead, if the surface is 0x0 semantics will just generate one root semantics node with size of 0x0. The node will have no children because all children are invisible and (except for the root node with this change) invisible children are dropped.

aam committed:
Roll engine to fed2ea458ed49088d33eddabc546ba56d600c717 (includes dart roll) (#19044)

* Roll engine to rolled dart
Add connectionTimeout property to class that extends HttpOverrides.
Upgrade packages.
Add more type fixes.
Update goldens hash.
Add more type fixes.
Upgrade packages.
Fix tests.
Remove strong-mode from analysis options yaml file.
Increase dev/devicelab timeout from 2 to 10 seconds.
Added getVM to waitForViews. Fix type errors in compileExpression.

* Pick up engine revision with fix for microbenchmarks regression.
Increase default timeout from 20/30 to 60s.

aam committed:
Revert "Roll engine to fed2ea458ed49088d33eddabc546ba56d600c717 (includes dart roll) (#19044)" (#19276)

This reverts commit f4a038da6328d437c9f0ca54d77ceb22779224a3 as it breaks microbenchmark tests.

Hixie commented:
cc @HansMuller for review since he's back.

Hixie committed:
Update style for CupertinoRefreshControl. (#19025)

All slivers must start with Sliver (or CupertinoSliver).
All widgets must have a key argument.

Also, some minor tweaks here and there to style and docs.

jslavitz committed:
Fixed code sign error message width (#19026)

* Fixed codesign error message width

* Fixed mac_test

* Removed snackbar code

xster committed:
Add a doc describing where CupertinoSliverRefreshControl should be in a list of slivers (#19282)

cbracken committed:
Push /usr/bin to front of PATH for ios-deploy runs (#19281)

ios-deploy relies on LLDB.framework, which relies on /usr/bin/python and
the 'six' module that's installed on the system. However, it appears to
use the first version of Python on PATH, rather than explicitly
specifying the system install.  If a user has a custom install of Python
(e.g., via Homebrew or MacPorts) ahead of the system Python on their
PATH, LLDB.framework will pick up that version instead. If the user
hasn't installed the 'six' module, ios-deploy will fail with a
relatively cryptic error message.

This patch pushes /usr/bin to the front of PATH for the duration of the
ios-deploy run to avoid this scenario.

This patch also removes checks for package six.

Neither Flutter nor any of its direct dependencies/tooling relies on
package six. ios-deploy depends on LLDB.framework (included with Xcode),
which relies on a Python script that imports this package but uses
whichever Python is at the front of the path. Flutter now invokes
ios-deploy with a PATH with /usr/bin forced to the front in order to
avoid this problem.

We could have retained the check out of paranoia, but this seems
unnecessary since it's entirely possible LLDB.framework may one day drop
this dependency, in which case I'd expect the base system install of
Python would likely drop it as well.

aam committed:
Roll engine to fed2ea458ed49088d33eddabc546ba56d600c717 (includes dart roll) again (#19277)

* Revert "Revert "Roll engine to fed2ea458ed49088d33eddabc546ba56d600c717 (includes dart roll) (#19044)" (#19276)"

This reverts commit cf932490b77c95841e8798e4864ac3e473a79e18 as it also
includes fix for type error that broke tests.

* Add type cast for dart2 type checks.

* Move up to latest goldens

* Make inDirectory() type-parameterized.

* Add typecasting to transitions_perf_test.dart and microbenchmarks.

* Add boolean flag initialization in save_catalog_screenshots.dart

* Add type conversion to gallery transition test

aam committed:
Revert "Roll engine to fed2ea458ed49088d33eddabc546ba56d600c717 (includes dart roll) again (#19277)" (#19295)

This reverts commit 700cb767c9b41c9db1686e9999776bc9f4126b8f as it
breaks Windows tests.

willlarche commented on packages/flutter/test/material/outline_button_test.dart:
```diff
@@ -43,90 +43,109 @@ void main() {
-              child: const Text('ABC')
-            ),
+            child:
+                new OutlineButton(onPressed: () {}, child: const Text('ABC')),
```
Still need `new` and `const` here in Dart 2?

Hixie replied:
Yes please explicitly specify all news and consts still.
We are working on having a lint that will help us consistently strip them out but we want the codebase to be consistent in the meantime.

willlarche commented on packages/flutter/test/material/outline_button_test.dart:
```diff
@@ -166,7 +184,7 @@ void main() {
             child: new Center(
               child: new OutlineButton(
-                onPressed: () { },
+                onPressed: () {},
```
Probably best to put all these style corrections in their own PR. NBD.

Commit:
Update translation descriptions: alertDialogLabel, searchFieldLabel (#19234)

DanTup committed:
Align expressions better (#19256)

DanTup committed:
Ensure timeouts in flutter_tools integration tests include all messages received (#19253)

* Allow a longer timeout for app launches

* Include timings in errors when waiting for events

* Extract a helper that adds a timeout and includes message logging

* Ensure all timeouts include messages

* Ensure logs are set up before we invoke actions

* Ensure all messages are logged during timeout periods

This adds a new stream _allMessages that collects all output (as routed through debugPrint) and updates _timeoutWithMessages to use that instead of stdin/out of only the flutter run process (since that was missing stdin - the sends, and both input/output to the VM service).

HansMuller commented:
Just FTR the goal here is to fix OutlineButton per https://github.com/flutter/flutter/issues/18170#issuecomment-402230741 by:
- Changing the outline border's width from 2.0 to 1.0
- Changing the outline border's color from Colors.grey[200] to Colors.black.withOpacity(0.12)

Current version:
![width2](https://user-images.githubusercontent.com/1377460/42655454-26cefccc-85d1-11e8-935b-4eb9bb79cd78.png)

New version:
![width1](https://user-images.githubusercontent.com/1377460/42655467-2b845c9e-85d1-11e8-9bfa-e678982ff960.png)

I assume that the outline's border width is 1.0 per the little "1" in the [material spec](https://material.io/design/components/buttons.html#specs):

![spec](https://user-images.githubusercontent.com/1377460/42655495-3b3e8ac4-85d1-11e8-8c79-95687edfbd56.png)

The spec doesn't appear to define the outline's color?

clocksmith commented:
The color can be seen if you click the button on bottom right of the spec
window, then select color in the right hand navigation:

It is #0000001e or r: 0 b: 0 g: 0 a:0.12

#0000001e

R0 G0 B0 A0.12

On Thu, Jul 12, 2018 at 3:48 PM Ian Hickson <notifications@github.com>
wrote:

> *@Hixie* commented on this pull request.
> ------------------------------
>
> In packages/flutter/test/material/outline_button_test.dart
> <https://github.com/flutter/flutter/pull/18929#discussion_r202157262>:
>
> >    testWidgets('OutlineButton contributes semantics', (WidgetTester tester) async {
>      final SemanticsTester semantics = new SemanticsTester(tester);
>      await tester.pumpWidget(
>        new Directionality(
>          textDirection: TextDirection.ltr,
>          child: new Material(
>            child: new Center(
> -            child: new OutlineButton(
> -              onPressed: () { },
> -              child: const Text('ABC')
> -            ),
> +            child:
> +                new OutlineButton(onPressed: () {}, child: const Text('ABC')),
>
> Yes please explicitly specify all news and consts still.
> We are working on having a lint that will help us consistently strip them
> out but we want the codebase to be consistent in the meantime.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/flutter/flutter/pull/18929#discussion_r202157262>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAkHU73nQoOGTRzCvCIuvkS1YOq4jxlWks5uF6gQgaJpZM4U8H9q>
> .
>

aam committed:
Preserve original kernel file for isolate spawn requests. (#18127)

* Preserve original kernel file for isolate spawn requests.

This change gives incremental kernel files ".incremental.dill" extension thereby preserving original kernel file.

Bug: https://github.com/flutter/flutter/issues/16084
Bug: https://github.com/flutter/flutter/issues/17012
Bug: https://github.com/flutter/flutter/issues/17983

* Factor out naming logic into separate function

* Fix string vs uri

zanderso committed:
Revert "Use runTests in fuchsia tester. (#19178)" (#19321)

This reverts commit 032f8cdb8b563bdc200acdce068945a834bf2271.

The test runner hangs on exit after all tests pass when trying
to roll this into Fuchsia.

aam committed:
Revert "Revert "Roll engine to fed2ea458ed49088d33eddabc546ba56d600c717 (includes dart roll) again (#19277)" (#19295)" (#19322)

This reverts commit cc122f1672d4b7e0c0734be34a155997463e3b56.

HansMuller commented:
@clocksmith Right, the spec says that the color is #0000001e. 

However we believe that it's really the (TBD) onSurfaceColor with 12% opacity.

Do we think the outline is actually supposed to be translucent, or is there another level of indirection here: should the outline color actually be an opaque color created by composing #0000001e with the default background color?

HansMuller commented:
The existing implementation, which seems to have been rewritten more than was needed, effectively defines 6 values:

```
configuration            widget override or null          current default
---------------------------------------------------------------------------
enabled pressed dark     widget.highlightedBorderColor    theme.primaryColor
enabled pressed light    widget.highlightedBorderColor    theme.primaryColor
enabled dark             widget.borderSide.color          Colors.grey[600]
enabled light            widget.borderSide.color          Colors.grey[200]
disabled dark            widget.disabledBorderColor       Colors.grey[800]
disabled light           widget.disabledBorderColor       Colors.grey[100]
```

Before updating OutlineButton we should understand exactly what the values in the default column should be and be able to point to where they're defined in the spec.

Commit:
Roll engine to c5a63d (#19288)

Commit:
Revert "Revert "Use runTests in fuchsia tester. (#19178)" (#19321)" (#19327)

This reverts commit e22f99743b4f0bea4a9b8e7703ce5ffbfda37a04.

Adding the exit(exitCode) fixes the hang.
Tested by running `fx run-host-tests` with a topaz checkout.
Verified the hang before, no hang after.

Commit:
Correct contentPadding type in InputDecoration.debugFillProperties (#19318)

Commit:
have text finder convert Text.rich to plain text for comparison (#19270)

xster committed:
Add HeroController to CupertinoApp (#19326)

Commit:
Add option to silence driver extension errors (#19247)

Commit:
Support keyboardAppearance field for iOS (#19244)

gspencergoog committed:
Add Bash and Zsh command-line completion for flutter (#19243)

This uses @kevmoo's completion package to do command line completion for flutter, and a new command "bash-completion" (with alias "zsh-completion") that will output the necessary shell script setup code, and adds the hidden command "completion" that does the actual completion.

Because it adds a dependency, I also had to do flutter update-packages --force-upgrade.

Fixes #18988.

Commit:
increase cache size if image is loaded that is larger than max size (#19352)

amirh committed:
roll engine to 9af920ef22f311588b649ba876127993f5c1362c (#19365)

amirh committed:
Revert "roll engine to 9af920ef22f311588b649ba876127993f5c1362c (#19365)" (#19376)

This reverts commit 44995dc42bc06ba9042f22707a3f9f78c21b7100.

sigurdm committed:
Use FlutterProject to locate files  (#18913)

sigurdm committed:
Fix typo (#19402)

sigurdm committed:
Extra debug information in run_release_test (#19405)

sigurdm committed:
Revert "Use FlutterProject to locate files  (#18913)" (#19409)

This reverts commit 57d78cc739a1d21a981886d424a6a104c8d48e33.

mjohnsullivan committed:
Chevrons in month picker are semi-transparent when the month is scrolled (#19363)

* Chevrons in month picker are semi-transparent when the month is scrolled

* Added type missing annotation

a14n committed:
enable lint unnecessary_const (#19342)

Hixie committed:
Revert "enable lint unnecessary_const (#19342)" (#19423)

This reverts commit cc1cf13eec6752af992d3db0265cf04cc90bfd13.

xster committed:
roll engine to 316b026807c50d65ba1b6724ac954711683f1b01 (#19419)

xster committed:
Revert engine roll to 316b026807c50d65ba1b6724ac954711683f1b01

This reverts commit b0eb1632d37feaf835d76d75108940cb2269d25b.

jason-simmons committed:
Fix return types of commands that override verifyThenRunCommand (#19483)

Fixes https://github.com/flutter/flutter/issues/19465

goderbauer committed:
Roll engine to 5557e3006e8346b2d3cccf3fa68859a00947641f (#19489)

rmacnak-google committed:
Build snapshots with --sync-async to match the engine. (#19495)

Bug: https://github.com/flutter/flutter/issues/19494

rmacnak-google committed:
Make initialSemanticsTreeCreation less sensitive to GC timing. (#19491)

Bug: https://github.com/flutter/flutter/issues/19435

goderbauer committed:
mark flaky benchmarks as flaky (#19492)

Hixie committed:
Clean up golden files (#19440)

Makes them all be generated on Linux and fixes some to not be so large.

zanderso committed:
[fuchsia] Add new package dep to BUILD.gn file (#19533)

gspencergoog committed:
Turn on MacOS tests on Cirrus (#19424)

Cirrus CI has macOS VMs, so this turns on an instance for tests so that we can evaluate if we can use Cirrus for all three platforms or not.

mjohnsullivan committed:
Chevrons fade out completely on month picker scroll event (#19422)

amirh committed:
Introduce PlatformViewsService for controlling platform views.  (#19325)

Currently only Android views are supported.

This also includes a FakePlatformViewsController which is a fake
implementation of the engine side for testing.

xster committed:
Fix sliver padding safe area for material gallery (#19497)

cbracken committed:
Eliminate Dart 1 benchmarks (#19430)

Flutter app development has been Dart 2 only for a significant amount of
time, and we're about to remove all support for Dart 1.

Commit:
Small update to the Text.rich API doc (#18721)

doxuanviet1996 committed:
Adding multicolumn cupertino picker to the Gallery. (#19284)

* Add sample of multicolumn picker to Flutter Gallery
* Modify CupertinoPicker to be able to be rendered off-center.

sigurdm committed:
Fix typo (#19555)

DanTup committed:
Flutter `emulator launch` crash running in dart-2-mode (#19454)

* Only printError on our simple messages

Any other type is a real error that should be a normal crash (to get a proper error log).

See #19453.

* Add toList() to convert Iterable<String> -> List<String>

This code previously throw in Dart-2 mode.

Fixes #19453.

* Move getSimulatorPath into Xcode

* Add a test that we tried to launch the simulator

* Remove unused import

DanTup committed:
Make cleanup of flutter processes in tests more reliable (#19307)

* Make cleanup of flutter processes in tests more reliable

* Fix quit signals

I confused SigInt&SigTerm for SigTerm&SigKill when I first did this. SigTerm can be blocked and doesn't guarantee the process will be terminated.

* Don't use deprecated constants

* Fix typo

* Add some additional info to debug buffer

* Fix return types on Futures

DanTup committed:
Add `--machine` support for `flutter attach` (#19077)

* Extract some of startApp into a reusable method

* Get basic attach --machine working

* Attach --machine tweaks

Move validation to validate method and create daemon early so we get the startup event before trying to get a connection.

* Bump daemon version so we know whether it's valid to flutter attach

* Tweak output text

* Swap package imports for relative

* Review tweaks (naming, formatting, typedefs)

* Separate arguments from process spawning

This will make calling attach easier

* Add a basic test for flutter attach --machine

* Fix crash if port unforward modifies the list of forwarded ports

* Add a no-op port forwarder for flutter-tester

* Switch to using BasicProject instead of our own inline code

* Fix expectation in test now we have a portForwarder

* Remove stale TODO (this is done)

* Tweak formatting

* Change some Completers to void to fix Dart 2 issues

DanTup committed:
Update devicelab test with new message prefix (#19560)

While working on attach, I changed the message output while waiting for an app to connect but didn't catch this.

Commit:
Enable extended FAB hero transition resizing (#19534)

cbracken committed:
Don't bundle Genereted.xcconfig as a resource (#19544)

Generated.xcconfig is only required at build time for iOS apps. In the
flutter create project template and example apps, Generated.xcconfig was
previously marked as a resource to be bundled into the built app.

amirh committed:
Skip flaky test (expression_evaluation_test) (#19543)

rmacnak-google committed:
Roll engine to 55b423f07ff747c2303c957fe151487b45585192.

55b423f07 Ensure assistiveTechnologyEnabled is initialized when the android view is set up (#5793)
3054f3183 Add touch events to the platform views method channel API. (#5796)
91c16af0c Revert "Roll Dart to 937ee2e8ca4b76499e24cd463f07bfb736bccd74. (#5745)" (#5799)
1a66f895b Roll Dart to 937ee2e8ca4b76499e24cd463f07bfb736bccd74. (#5745)
8d046a727 Revert "Roll src/third_party/skia 9e0d7e4072e4..8f8bf8880d9d (32 commits) (#5794)" (#5797)
e8d172cf3 Revert "Fix Dart 2 reload when running from a snapshot instead of platform.dill. (#5792)" (#5795)
624cf7e25 Fix Dart 2 reload when running from a snapshot instead of platform.dill. (#5792)
16da4710d Roll src/third_party/skia 9e0d7e4072e4..8f8bf8880d9d (32 commits) (#5794)
3fe63fd5d Add presubmit check for engine to flutter roll (#5790)

rmacnak-google committed:
Roll engine to 09ff9493e08c0c937bcd2c6f780981e2a7c5f2b5.

09ff9493e Re-land removal of vmservice_io from release builds. (#5784)
b523ee847 Revert "Switch to Fuchsia's ICU (#5774)" (#5801)
69695493b Switch to Fuchsia's ICU (#5774)

rmacnak-google committed:
Roll engine to 5d0e13a45b210b9b05427cccfc5f94928ae701ab.

5d0e13a45 Fix starting isolate directly from CoreJIT snapshot. (#5798)

rmacnak-google committed:
Roll engine to 482469bad8094ff05fd66dfc07149fbef4ae00ac.

482469bad Roll Dart to 937ee2e8ca4b76499e24cd463f07bfb736bccd74. (#5800)
bb323c145 Revert "Roll src/third_party/skia 9e0d7e4072e4..297c3c8e85bd (62 commits) (#5802)" (#5803)
0bf69f02d Roll src/third_party/skia 9e0d7e4072e4..297c3c8e85bd (62 commits) (#5802)

DanTup committed:
Fix copy/paste error in test (#19564)

tianlunlee committed:
Add build test for Flutter codelabs

jason-simmons committed:
Keep the debug repaint rainbow color within a valid range (#19355)

Fixes https://github.com/flutter/flutter/issues/18971

tianlunlee committed:
Revert accidental force push: 99457c92d78090759166b2710ba2bf4d2faa122e (#19573)

matthew-carroll committed:
Added textInputAction parameter to TextFormField that forwards to TextField, added onEditingComplete parameter to TextField that forwards to EditableText, added onEditingComplete parameter to TextFormField that forwards to TextField. (#19397) (#19427)

Commit:
Added 20 new supported locales to Material (#19567)

Commit:
Allow material button to grow as wide as its constraints allow (#19416)

Commit:
add textCapitalization property (#19367)

eyallen committed:
Add optional haptic feedback on LongPressDraggable (#18781)

Commit:
Add support for custom semantics actions to Android and iOS. (#18882)

sigurdm committed:
Revert "Revert "Use FlutterProject to locate files (#18913)" (#19409)"  (#19456)

With a fix of a path being printed relative instead of absolute.

gspencergoog committed:
Removes trailing whitespace from source files. (#19329)

gspencergoog committed:
Add presubmit test for trailing whitespace. (#19412)

Checks only changed source files (C++, Dart, Java, ObjC) for trailing whitespace.

szakarias committed:
add more uses of FlutterProject to locate files (#19403)

Norbert515 committed:
Making the space between keyboard and text-field when the keyboard pops up a variable. (#18562)

gspencergoog committed:
Separate tools tests from other tests and put them in a different shard. (#19593)

goderbauer committed:
Whitespace fixes to make Travis happy (#19611)

Follow-up to https://github.com/flutter/flutter/pull/18562, which passed Travis before the check was added.

Hixie committed:
Increase verbosity on bots during setup and other minor cleanup (#19526)

The verbosity change is to help track down timeouts that currently look like:

```
C:\Windows\Temp\flutter sdk>call bin\flutter.bat config --no-analytics
Checking Dart SDK version...
Downloading Dart SDK from Flutter engine c5a63d28bf3735569c8187753bc490d8351a8363...
Unzipping Dart SDK...
Updating flutter tool...
```

leafpetersen committed:
Remove uses of deprecated constants and change int.parse to int.tryParse (#19575)

* Remove uses of deprecated constants
* Change int.parse to int.tryParse where appropriate

jslavitz committed:
Changed duration (#19015)

* Changed duration

* Fixed column width for ios build error messages

* fixed a couple things

* fixed a couple things

* accidental changes

* added spaces

* added spaces

* Added test for default snackbar duration

* Fixed link:

* Final commit

* Final commit2

* Actual final commit

amirh committed:
Introduce an AndroidView widget and a RenderAndroidView render object. (#19565)

RenderAndroidView is responsible for sizing and displaying an embedded
Android view.
AndroidView is responsible for creating and disposing the Android view
and is using RenderAndroidView to display it.

gspencergoog committed:
Shard the Appveyor tests. (#19631)

Commit:
Roll engine to 96f5f5b96c7220ffc8c4ce22ed895616a15ad3a0 (#19615)

Commit:
Change material RefreshIndictor.onRefresh signature from Future<Null> to Future<void> (#18775)

* Change material refresh indicator onRefresh signature from Future<Null> to Future<void>.

* Update authors.

* Explicitly name void argument.

a14n committed:
re-enable lint unnecessary_const (#19592)

* re-enable lint unnecessary_const

* remove trailling whitespaces

* remove unnecessary const (after merge)

Commit:
Remove redundant check for platform.isLinux. (#19441)

We don't actually care if we're on linux, we care only if the lcov
tool is available. We check for that explicitly just below.
Additionally there is code below which indicates that lcov is available
for macOS. Before this change that code would never execute.

DanTup committed:
In microbenchmarks, ensure flutter quits by sending `q` to stdin (#19400)

* Tell flutter to quit directly

Fixes #19208.

* Tweak comment

Hixie committed:
Fix the checksum verification to be more sensitive (#19290)

* Fix the checksum verification to be more sensitive

* update packages

Commit:
Add ignoringSemantics argument to AbsorbPointer (#19651)

gspencergoog committed:
Fix whitespace detector to handle deleted files. (#19690)

The trailing whitespace detector wasn't handling file deletes very well (at all, really).

This filters the set of files grepped to only include files that exist.

Also, clarified the failure message to make it more obvious what the failure is when the grep finds results.

sandrasandeep committed:
Add cursorWidth and cursorRadius to TextField (Material) cursor (#19317)

* fixed segmented control golden test

* fixed segmented control golden test

* made the comments more explanatory

* changed comment, wrapped tests with if statements so they only run on MacOS

* changed formatting

* added width and radius fields to TextField; to do: tests and Material defaults

* weak warnings

* added tests

* added default cursor width; changed default EditableText width

* only run golden file tests on linux

* changed goldens version

* actually changed goldens.version

* style changes

* small fixes

* added default material cursor color

* changed goldens.version

* changed goldens version again

* changed goldens.version again (3)

* added todo

* deleted whitespace

DanTup committed:
Run "flutter update-packages --force-upgrade" (#19679)

Hixie committed:
Improve docs for FutureBuilder (#19687)

Commit:
Extend InputDecoration borders for disabled/error/focused states (#19694)

Added five InputBorder properties to InputDecoration etc so that apps can control the appearance of an InputDecorator's border:  errorBorder, focusedBorder, focusedErrorBorder, disabledBorder, enabledBorder.

matthew-carroll committed:
Fix Flutter.java initialization crash (#19628)

Commit:
Semantics framework updates (#18758)

Changes:

- Move the SemanticsConfiguration update from RenderToggleable to each subclass, so that Switch can use toggled.
- Add image, liveRegion, toggled properties to Semantics, SemanticsConfiguration, SemanticsNode
- Added semanticsLabel and excludeFromSemantics to Image (the latter so that we avoid creating a semantics node)
- Added onDismiss semantics action which maps to the modal escape on iOS and dismiss action on Android.
- Added dismiss and liveRegion to snackbar widget
- Updated custom painter semantics to handle image, liveRegion, toggle
- Updated relevant tests to use correct flag/action

liyuqian committed:
Roll engine to 95e91e31fad816738d3fef341256620e1819b6d9 (#19716)

95e91e31f Update tonic to pull in Android build fixes. (#5834)
29f475c8d Update tonic to pull in fixes to assertions and build failures on Windows. (#5833)
5474415e1 Roll src/third_party/skia e35055f31c7f..b8eeb808d84f (5 commits) (#5832)
380eb84c4 Remove unused function CreateSkiaInterface (#5812)
33a4aef1d Fix build break caused by https://github.com/flutter/engine/pull/5830 (#5831)
244ebd660 Reapply "Fix Dart 2 reload when running from a snapshot instead of platform.dill." (#5830)
0bad812c7 Migrate to a standalone Tonic repository separated from Topaz (#5817)
f949f7d43 Roll src/third_party/skia 206dbe8a21b6..e35055f31c7f (1 commits) (#5829)
768daa4d3 Roll src/third_party/skia 5e6cd2affe41..206dbe8a21b6 (5 commits) (#5828)
b8cd25b4f libtxt: fix a bug that used incorrect buffer size depending on conditions (#5804)
b7b13b5e9 Roll src/third_party/skia df795638b9bd..5e6cd2affe41 (1 commits) (#5824)
5886949c9 Roll src/third_party/skia dfe1fe40796e..df795638b9bd (1 commits) (#5822)
926f546de Roll src/third_party/skia cc22d4d2fb02..dfe1fe40796e (1 commits) (#5821)
04633f326 Generate process global trace identifiers for pipeline flow traces. (#5818)
fb0bc05d7 Roll src/third_party/skia 6f6de6524c73..cc22d4d2fb02 (1 commits) (#5820)
fadfff7fd Mention that running git clone isn't necessary (#5816)
81ec6000d Roll src/third_party/skia d3509a42638a..6f6de6524c73 (5 commits) (#5819)
2a77a4130 Make git diff range more robust in format.sh (#5813)
e555ea573 Remove trailing white spaces (#5734)
4c7618ac7 Roll src/third_party/skia fa0720436319..d3509a42638a (7 commits) (#5814)

Commit:
Fix off by one error in TextPainter. (#19429)

Commit:
remove trailing whitespace (#19728)

rmacnak-google committed:
Revert "Make initialSemanticsTreeCreation less sensitive to GC timing. (#19491)"

This reverts commit 3840719987c4b42e7b8a5557c3b47de73bae400f.

Bug: https://github.com/flutter/flutter/issues/19435

tvolkert committed:
Make Fuchsia tools Dart 2 clean (#19727)

moznion committed:
Add some options to format subcommand (#18360)

* Add `--dry-run` option to `flutter format` sub command

* Add `--set-exit-if-changed` option to `format` sub command

* Add `--machine` option to `format` sub command

* Make variable names to be not shorthand: cmd -> command

Fix https://github.com/flutter/flutter/pull/18360#discussion_r199656120

jaggernod committed:
Added SM-G960F to the list of known physical devices (#18966)

Samsung Galaxy S9 is not an emulator and installation of
the release packages with 'flutter run --release' should be possible.

DanTup committed:
Rename file -> _test so it runs on CI! (#19749)

DanTup committed:
Skip attach test on Windows due to #17833 (#19763)

DanTup committed:
Look up breakpoints rather than relying on magic numbers (#19708)

* Look up breakpoints rather than relying on magic numbers

* Make breakpoint comment more obvious

DanTup committed:
Add integration tests to ensure flutter run does not quit at startup (#19473)

* Add tests to ensure flutter run does not quit at startup

These tests require https://github.com/dart-lang/vm_service_client/pull/38 to be merged and currently the third test (pause-on-exceptions) fails because of a bug.

* Remove TODO

Probably makes most sense this way for now.

* Address PR comments

DanTup committed:
Skip integration tests that use flutter-tester on Windows (#19773)

* Skip integration tests that use flutter-tester on Windows

This is known to be broken. I will un-skip all of these once a fix for flutter-tester lands on Windows.

* Save with added import!

gspencergoog committed:
Enhance SizeTransition docs (#19721)

Updated documentation of SizeTransition, since I found it less than intuitive.

I added some constructor asserts to make the (existing) contract more clear.

gspencergoog committed:
Empty commit to force bots to start.

amirh committed:
Fix touch event conversion bugs. (#19780)

These are bugs uncovered by the integration test for embedded Android
views:

  * Some PointerEvents were constructed with radiusMinor set to
  radiusMajor.
  * Some PointerEvents were missing radiusMajor/radiusMinor.
  * The pressure parameter was not passed to PointerUpEvent.
  * radiusMin, radiusMax, radiusMinor, and radiusMajor were not
  converted from physical pixels to logical.

Sending the fixes without a test as the bugs are covered by the integration
test that is going to land soon.

Rahiche committed:
Update README.md (#19717)

the thumbnail is not related to the link anymore 
https://github.com/flutter/flutter/issues/17380

Commit:
Add excludeSemantics flag to semantics widget (#19650)

jslavitz committed:
Added single open panel functionality (#19624)

* Just commiting two files now

* Fixed one tab

* Fixed latest changes

* Changed the data structure from a map to a single radio panel object

* A few more changes

* Fixed change from expansion radio to regular list

* Fixed change from expansion radio to regular list2

* Changed the radio constructor

* Last fixes

* Final commit

* Actual final commit

* Last change

gspencergoog committed:
Move commit range determination inside of test.dart (#19777)

Our Cirrus Windows tests were not really working: they reported success even when failing.

This should fix several problems with the Cirrus Windows tests.

amirh committed:
Roll engine to 7d17da76d (#19785)

```
7d17da76d Package Skia licenses into a separate file from other third party libraries (#5861)
d2d8bf394 Pass touch events to embedded Android views with dispatchTouchEvents.
216908652 Synthesize Android MotionEvents with the long form obtain method.
0b4432769 Roll src/third_party/skia 396661825f45..1c94a8fabed7 (8 commits) (#5859)
f50e218d9 Rename clip mode to clip behavior (#5853)
42bd86d6d Make the virtual display's presentation window non focusable. (#5845)
a726e1d85 Roll src/third_party/skia c062b6be5a01..396661825f45 (3 commits) (#5857)
3fa83d2d2 Roll src/third_party/skia 15020ea2cf44..c062b6be5a01 (1 commits) (#5856)
75ad43954 Roll src/third_party/skia 4c7a750e5543..15020ea2cf44 (1 commits) (#5855)
5cd8b4530 Roll src/third_party/skia 75e5406be66a..4c7a750e5543 (2 commits) (#5854)
e12b52841 Roll src/third_party/skia b8eeb808d84f..75e5406be66a (17 commits) (#5852)
5bc1e68b6 Remove check_roll from Travis (#5846)
```

Commit:
Revert "Roll engine to 7d17da76d (#19785)" (#19792)

This reverts commit 8adecb62e1701decdf22d76ddec663b2d5f7743e.

stevemessick committed:
Add IJ module definitions for modules (#19722)

amirh committed:
Revert "Revert "Roll engine to 7d17da76d"" (#19825)

* Revert "Revert "Roll engine to 7d17da76d (#19785)" (#19792)"

This reverts commit e1a3e1bfd8c52153af01f1e6f361f5f398e92862.

leafpetersen committed:
Roll pubspec forward to new json_schema, matcher and vector_math (#19802)

Pull in new versions of packages with fixes for SCREAMING CAPS CONSTANTS.

Commit:
Add header SemanticsFlag to bottom nav bar (#19803)

Hixie committed:
Clean up startProgress logic. (#19695)

Disallow calling stop() or cancel() multiple times. This means that
when you use startProgress you have to more carefully think about what
exactly is going on.

Properly cancel startProgress in non-ANSI situations, so that
back-to-back startProgress calls all render to the console.

Commit:
Revert "Clean up startProgress logic. (#19695)" (#19842)

This reverts commit 0636c6fe6026c24763738886db94ba948e7fd5a5.

Commit:
Make Card explicitChildNodes vs container be configurable (#19693)

tvolkert committed:
More dart 2 fixes to fuchsia_tester.dart (#19846)

creativecreatorormaybenot committed:
Added 35 icons (#19725)

Commit:
handle case where image is null as well (#19797)

Commit:
roll engine to 6f7bdc9fabef810b14577deef5cf330029fd290e (#19848)

tvolkert committed:
Revert flutter/flutter#19592 (#19861)

It was causing problems rolling Flutter into Fuchsia

tvolkert committed:
Revert "roll engine to 6f7bdc9fabef810b14577deef5cf330029fd290e" (#19878)

* Revert "Revert flutter/flutter#19592 (#19861)"

This reverts commit 00aac68e2da653920aac8c82d52b9382d3743895.

* Revert "roll engine to 6f7bdc9fabef810b14577deef5cf330029fd290e (#19848)"

This reverts commit 8b05629658c17011394451b0997433d7068d719e.

tvolkert committed:
Revert "Revert "roll engine to 6f7bdc9fabef810b14577deef5cf330029fd290e" (#19878)" (#19880)

This reverts commit e52eeba3674b1c5cc6ee46c1088f1f0b997a311c.

clocksmith committed:
Merge branch 'fix-outlined-button-stroke' of https://github.com/clocksmith/flutter into fix-outlined-button-stroke
